soh/soh/Enhancements/randomizer/randomizer_entrance_tracker.cpp

```diff
@@ -0,0 +1,877 @@
+        return;
+    }
+
+    // Set total and group counts
+    for (size_t i = 0; i < ENTRANCE_OVERRIDES_MAX_COUNT; i++) {
+        if (Entrance_EntranceIsNull(&gSaveContext.entranceOverrides[i])) {
+            break;
+        }
+        const EntranceData* index = GetEntranceData(gSaveContext.entranceOverrides[i].index);
+        const EntranceData* override = GetEntranceData(gSaveContext.entranceOverrides[i].override);
+
+        if (index->srcGroup == ENTRANCE_GROUP_ONE_WAY) {
+            gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_SOURCE_AREA][ENTRANCE_GROUP_ONE_WAY]++;
+            gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_DESTINATION_AREA][ENTRANCE_GROUP_ONE_WAY]++;
+            gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_SOURCE_TYPE][ENTRANCE_TYPE_ONE_WAY]++;
+            gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_DESTINATION_TYPE][ENTRANCE_TYPE_ONE_WAY]++;
+        } else {
+            gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_SOURCE_AREA][index->srcGroup]++;
+            gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_DESTINATION_AREA][override->dstGroup]++;
+            gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_SOURCE_TYPE][index->type]++;
+            gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_DESTINATION_TYPE][override->type]++;
+        }
+        gEntranceTrackingData.EntranceCount++;
+    }
+
+    // Set offset
+    u16 srcOffsetTotal = 0;
+    u16 dstOffsetTotal = 0;
+    for (size_t i = 0; i < SPOILER_ENTRANCE_GROUP_COUNT; i++) {
+        gEntranceTrackingData.GroupOffsets[ENTRANCE_SOURCE_AREA][i] = srcOffsetTotal;
+        srcOffsetTotal += gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_SOURCE_AREA][i];
+        gEntranceTrackingData.GroupOffsets[ENTRANCE_DESTINATION_AREA][i] = dstOffsetTotal;
+        dstOffsetTotal += gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_DESTINATION_AREA][i];
+    }
+    srcOffsetTotal = 0;
+    dstOffsetTotal = 0;
+    for (size_t i = 0; i < ENTRANCE_TYPE_COUNT; i++) {
+        gEntranceTrackingData.GroupOffsets[ENTRANCE_SOURCE_TYPE][i] = srcOffsetTotal;
+        srcOffsetTotal += gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_SOURCE_TYPE][i];
+        gEntranceTrackingData.GroupOffsets[ENTRANCE_DESTINATION_TYPE][i] = dstOffsetTotal;
+        dstOffsetTotal += gEntranceTrackingData.GroupEntranceCounts[ENTRANCE_DESTINATION_TYPE][i];
+    }
+
+    // Sort entrances by group and type in entranceData
+    for (size_t i = 0; i < ENTRANCE_OVERRIDES_MAX_COUNT; i++) {
+        srcList[i] = gSaveContext.entranceOverrides[i];
+        destList[i] = gSaveContext.entranceOverrides[i];
+        srcTypeList[i] = gSaveContext.entranceOverrides[i];
+        destTypeList[i] = gSaveContext.entranceOverrides[i];
+    }
+    SortEntranceList(srcList, 0);
+    SortEntranceList(destList, 1);
+    SortEntranceListByType(srcTypeList, 0);
+    SortEntranceListByType(destTypeList, 1);
```

Comment:
ah, i think i understand why those have the same values, might be good to rename those to better represent what they are, a list of every entrance that will be sorted by type (maybe `typeSortedEntranceSrcList` or something?)